cartographer/mapping_2d/sparse_pose_graph.cc

```diff
@@ -114,10 +116,14 @@ void SparsePoseGraph::AddScan(
     submap_states_.back().trajectory = submaps;
     CHECK_EQ(submap_states_.size(), submap_indices_.size());
   }
+  // ... One call before the first of insertion_submaps was finished
```

Comment:
Here we figure out whether a submap was finished. In practice this might happen at a similar time like growing `submap_states_` (the code above), but is conceptually two separate concerns. I would suggest not adding comments that tie these concerns together. This would encode the current implementation details in comments.